mapsforge-map-writer/src/main/java/org/mapsforge/map/writer/util/PolyLabel.java

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright 2016 Andrey Novikov
+ * Java implementation and adoption for JTS and MapsForge of
+ * https://github.com/mapbox/polylabel
+ * <p>
+ * ISC License
+ * Copyright (c) 2016 Mapbox
+ * <p>
+ * Permission to use, copy, modify, and/or distribute this software for any purpose
+ * with or without fee is hereby granted, provided that the above copyright notice
+ * and this permission notice appear in all copies.
+ * <p>
+ * THE SOFTWARE IS PROVIDED "AS IS" AND ISC DISCLAIMS ALL WARRANTIES WITH REGARD TO
+ * THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS.
+ * IN NO EVENT SHALL ISC BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
+ * CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA
+ * OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+ * ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ */
+package org.mapsforge.map.writer.util;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.CoordinateFilter;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.Polygon;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+import java.util.logging.Logger;
+
+public class PolyLabel {
+    private static final Logger LOGGER = Logger.getLogger(PolyLabel.class.getName());
+    private static final float SQRT2 = (float) Math.sqrt(2);
```

Comment:
I was writing the code with floats first and then switched to doubles because JTS operates with doubles. Should be `double SQRT2 = Math.sqrt(2.0)` of course.